### PR TITLE
Issue #1254: Fix JavadocVariableCheck when assigning a lambda as a field

### PR DESCRIPTION
**Pull request by MEZk**
#1254

Diff reoports were generated against the following projects:
1. openjdk8
2. pmd
3. lombok-ast
4. spring-framework
5. elasticsearch
6. java-design-patterns
7. MaterialDesignLibrary
8. Hbase
9. Orekit
10. apache-ant
11. apache-jsecurity
12. android-launcher
13. apache-struts
14. infinispan
15. protonpack
16. jOOL
17. RxJava
18. checkstyle-with-excludes
19. sevntu-checkstyle-with-excludes
20. findbugs
21. hibernate-orm-with-excludes
22. guava-mvnstyle

http://mezk.github.io/i1254-JavadocVariable/index.html

### CODE REVIEW

**Comment by rnveach** [comment]
Looks like travis was cancelled. I restarted the job.

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtils.java [hunk 1]
```diff
@@ -209,7 +209,8 @@ public static boolean isInCodeBlock(DetailAST node) {
             if (type == TokenTypes.METHOD_DEF
                 || type == TokenTypes.CTOR_DEF
                 || type == TokenTypes.INSTANCE_INIT
-                || type == TokenTypes.STATIC_INIT) {
+                || type == TokenTypes.STATIC_INIT
+                || type == TokenTypes.LAMBDA) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtils.java]: Why lambda and not field definition? Are all lambdas considered a code block from the perspective of this utility?
There is no removal of violation in report for this which is still a lambda: https://mezk.github.io/i1254-JavadocVariable/xref/home/aselkin/IDEA_Projects/contribution/checkstyle-tester/src/main/java/openjdk8/src/share/classes/java/time/temporal/TemporalQueries.java.html#L381
Makes me think lambda isn't the direct issue.

Either way, how about we add the change to the method javadoc? It wasn't updated.

**Comment by MEZk** [comment]
@rnveach 
>One concern: You changed a common utility but only ran regression against 1 check. If @romani agrees, I think regression should be run against all checks that use this method to be sure of impact.
Other checks include: MagicNumber, InnerTypeLast, JavadocStyle, JavadocVariable, ConstantName

Agree, I'll provide regression reports.

>Are all lambdas considered a code block from the perspective of this utility?
Yes.

>There is no removal of violation in report for this line which is still a lambda: 

The issues is with the javadocs for local variables inside lambda expression. Without a fix the check requires a javadoc to be placed inside lambda for local variables. Javadoc in your example is for lambda itself.

See example of wrong behavior: https://github.com/checkstyle/checkstyle/issues/1254#issue-91600982

**Comment by MEZk** [comment]
@rnveach 

Diff reports which were generated against the projects mentioned in my first comment (1 - 22) in order to check all checks that use isInCodeBlock method.

http://mezk.github.io/i1254-JavadocVariable/isInCodeBlock-diff/index.html

**Comment by rnveach** [comment]
@MEZk Thanks. No new differences.
[Please update the Javadoc and I'll approve the changes. ](https://github.com/checkstyle/checkstyle/pull/3520#discussion_r86235033)

**Comment by MEZk** [comment]
@rnveach 
>Please update the Javadoc and I'll approve the changes. 

Done.

**Comment by romani** [comment]
nothing to comment, good job!

**Pull request merged**